roles/ssh_role/tasks/main.yml

```diff
@@ -0,0 +1,46 @@
+---
+# tasks file for ssh_role
+- name: Set authorized key on rpi taken from local file
+  ansible.posix.authorized_key:
+    user: "{{ ansible_user }}"
+    state: present
+    key: "{{ lookup('file', public_key_path) }}"
+
+- name: Generated ssh key on rpi
+  openssh_keypair:
+    path: ~/.ssh/id_rsa
+    type: rsa
+
+- name: fetch rpi public key to copy it on local host
+  ansible.builtin.fetch:
+    src: ~/.ssh/id_rsa.pub
+    dest: ../files/ssh_rpi_keys/id_rsa_{{ inventory_hostname }}.pub
+    flat: true
+
+- name: fetch rpi ecdsa key to copy it on local host
+  ansible.builtin.fetch:
+    src: /etc/ssh/ssh_host_ecdsa_key.pub
+    dest: ../files/ssh_ecdsa_keys/ssh_host_ecdsa_key_{{ inventory_hostname }}.pub
+    flat: true
+
+- name: list file in directory
+  ansible.builtin.find:
+    paths: ../files/ssh_rpi_keys
+  register: files_list
+  delegate_to: 127.0.0.1
+
+- name: set authorized_keys on rpi
+  ansible.posix.authorized_key:
+    user: "{{ ansible_user }}"
+    state: present
+    key: "{{ lookup('file', item.path ) }}"
+  with_items: "{{ files_list.files }}"
+
+- name: add rpi public key in known_hosts for all rpis
```

Comment:
sorry, maybe I missed the logic: does it add the keys of all rpis as authorized keys to each of them?

Reply:
This tasks adds the ecdsa keys of all rpis as known_hosts to each of them (this then allows each rpi to connect to all others. It is for ingoing connexion). The known_hosts file tells the rpi where it can go.

Whereas the task "set authorized_keys on rpi" adds the rsa keys of all rpis as authorized_keys to each of them (to allow all other rpis to connect to the other; this for outgoing connexion). The authorized_keys file tells the rpi which other rpi can come and ssh on it.

Reply:
Maybe we can change the names of the tasks to make it more clear

Reply:
Sure, changing the name could clarify!